packages/multichain/src/caip25Permission.ts

```diff
@@ -0,0 +1,283 @@
+import type { NetworkClientId } from '@metamask/network-controller';
+import type {
+  PermissionSpecificationBuilder,
+  EndowmentGetterParams,
+  ValidPermissionSpecification,
+  PermissionValidatorConstraint,
+  PermissionConstraint,
+} from '@metamask/permission-controller';
+import {
+  CaveatMutatorOperation,
+  PermissionType,
+} from '@metamask/permission-controller';
+import type { CaipAccountId, Json } from '@metamask/utils';
+import {
+  parseCaipAccountId,
+  type Hex,
+  type NonEmptyArray,
+} from '@metamask/utils';
+import { cloneDeep, isEqual } from 'lodash';
+
+import { getEthAccounts } from './adapters/caip-permission-adapter-eth-accounts';
+import { assertScopesSupported } from './scope/assert';
+import { validateAndNormalizeScopes } from './scope/authorization';
+import type {
+  ExternalScopeString,
+  InternalScopeObject,
+  InternalScopesObject,
+} from './scope/types';
+
+export type Caip25CaveatValue = {
+  requiredScopes: InternalScopesObject;
+  optionalScopes: InternalScopesObject;
+  sessionProperties?: Record<string, Json>;
+  isMultichainOrigin: boolean;
```

Comment:
Do we still need this? Given [this discussion](https://consensys.slack.com/archives/C04M4QU03MG/p1728414138186019?thread_ts=1721941549.941049&cid=C04M4QU03MG)

Reply:
I guess we should keep it just in case we need different UI for these connections. Though I am not sure if we need this for the UI either.

Reply:
Is it okay if we punt this question to the next PR?